File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/producer/inject/BookSender.kt
package io.micronaut.kafka.docs.producer.inject

// tag::imports[]
import io.micronaut.configuration.kafka.annotation.KafkaClient
import io.micronaut.context.annotation.Requires
import jakarta.inject.Singleton
import org.apache.kafka.clients.producer.Producer
import org.apache.kafka.clients.producer.ProducerRecord
import org.apache.kafka.clients.producer.RecordMetadata
import java.util.concurrent.Future
// end::imports[]

// tag::clazz[]
@Requires(property = "spec.name", value = "BookSenderTest")
@Singleton
class BookSender(
    @param:KafkaClient("book-producer") private val kafkaProducer: Producer<String, Book>) { // <1>

    fun send(author: String, book: Book): Future<RecordMetadata> {
        return kafkaProducer.send(ProducerRecord("books", author, book)) // <2>
    }
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/kafka/docs/producer/inject/BookSender.py
# tag::imports[]
from typing import Annotated

from jakarta.inject import Singleton
from java.util.concurrent import Future
from micronaut.configuration.kafka.annotation import KafkaClient
from micronaut.context.annotation import Requires
from org.apache.kafka.clients.producer import Producer, ProducerRecord, RecordMetadata

from .Book import Book
# end::imports[]


@Requires(property="spec.name", value="BookSenderTest")
# tag::clazz[]
@Singleton
class BookSender:

    def __init__(self, kafka_producer: Annotated[Producer[str, Book], KafkaClient("book-producer")]):  # <1>
        self.kafka_producer = kafka_producer

    def send(self, author: str, book: Book) -> Future[RecordMetadata]:
        return self.kafka_producer.send(ProducerRecord("books", author, book))  # <2>
# end::clazz[]
